File: examples/maven-spring-compile/src/main/java/community/flock/wirespec/example/gradle/app/todo/LiveTodoRepository.java
package community.flock.wirespec.example.gradle.app.todo;

import community.flock.wirespec.generated.java.Todo;
import community.flock.wirespec.generated.java.TodoId;
import org.springframework.stereotype.Repository;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static java.util.Collections.emptyList;

@Repository
public class LiveTodoRepository implements TodoRepository {

    private static final String UUID = "f20ad876-c6a8-48b8-9a23-71787c1ae34a";
    private final HashMap<String, Todo> todos = new HashMap<>(
            Map.of(
                    UUID,
                    new Todo(
                            new TodoId(UUID),
                            "Name",
                            true,
                            emptyList()
                    )
            )
    );

    @Override
    public List<Todo> getAllTodos() {
        return todos.values().stream().toList();
    }

    @Override
    public Todo getTodoById(final TodoId id) {
        return todos.get(id.value());
    }

    @Override
    public Todo saveTodo(final Todo todo) {
        todos.put(todo.id().value(), todo);
        return todo;
    }

    @Override
    public Todo deleteTodoById(final TodoId id) {
        return todos.remove(id.value());
    }

}

File: examples/wirespec-gradle-plugin-ktor/src/main/kotlin/community/flock/wirespec/examples/app/todo/LiveTodoRepository.kt
package community.flock.wirespec.examples.app.todo

import community.flock.wirespec.examples.app.exception.TodoNotFoundException

class LiveTodoRepository : TodoRepository {

    private val todos = mutableListOf(
        Todo(
            id = Todo.Id("8132b795-143f-4afb-8c8a-0608cb63c79c"),
            name = Name("Name"),
            done = true,
        )
    )

    override fun getAllTodos(): List<Todo> = todos

    override fun getTodoById(id: Todo.Id): Todo = todos.find { it.id == id } ?: throw TodoNotFoundException(id)

    override fun saveTodo(todo: Todo): Todo = todo.also(todos::add)

    override fun deleteTodoById(id: Todo.Id): Todo = getTodoById(id).also(todos::remove)

}
